pkgs/development/libraries/alice-vision/3.0.0.nix

```diff
@@ -0,0 +1,225 @@
+, openimageio
+, zlib
+
+, enableOpenMP ? true
+# Separate openmp input only required on Darwin stdenv (LLVM)
+, openmp
+
+, enableAlembic ? true, alembic
+, enableCctag ? true, cctag
+, enableOpenCV ? true, enableOpenCVContrib ? enableOpenCV, opencv
+}:
+
+let
+  nanoflann = pkgs.nanoflann.overrideAttrs (
+    oldAttrs: rec {
+      version = "1.4.2";
+      src = fetchFromGitHub {
+        owner = "jlblancoc";
+        repo = "nanoflann";
+        rev = "v${version}";
+        hash = "sha256-znIX1S0mfOqLYPIcyVziUM1asBjENPEAdafLud1CfFI=";
+      };
+    }
+  );
+in
+stdenv.mkDerivation rec {
+  pname = "alice-vision";
+  version = "3.0.0";
+
+  outputs = [ "out" "dev" ];
+
+  src = fetchFromGitHub {
+    owner = "alicevision";
+    repo = "AliceVision";
+    rev = "v${version}";
+    hash = "sha256-rFd2AFtvC1RugKGv4tI2k3rtDqXHjdDC3pgOrqDqIT0=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    flann
+    nanoflann
+    openexr
+
+    # Temporary fix until flann 1.9.2 is in Nixpkgs
+    lz4
+  ];
+
+  propagatedBuildInputs = [
+    assimp
+    boost179
+    ceres-solver
+    clp
+    eigen
+    expat
+    geogram
+    lemon-graph
+    openimageio
+    zlib
+  ] ++ lib.optional enableAlembic alembic
+    ++ lib.optional enableCctag cctag
+    ++ lib.optional enableOpenCV opencv
+    ++ lib.optional stdenv.cc.isClang openmp;
+
+  patches = [
+    # Don't optimize for the host processor (could lead to reproducibility issues)
+    ./cmake-disable-optimize-for-architecture.patch
+
+    # Upstream uses patched Clp/Osi/CoinUtils containing CMake build scripts.
+    # Instead of the patched versions, we re-use our packages and add CMake
+    # find modules that use the pkg-config files already generated by those
+    # dependencies.
+    ./find-coin-modules.patch
+
+    ./find-flann.patch
+  ];
+
+  # Instead of using dependencies from Git submodules, we use Nix packages
+  # This speeds up fetching and reduces the source archive size
+  postPatch = ''
+    rmdir src/dependencies/nanoflann
+    ln -s ${nanoflann} src/dependencies/nanoflann
+
+    rm -r src/dependencies/lemon
+    rm -r src/dependencies/flann
+
+    rm src/cmake/FindFlann.cmake
+
+    substituteInPlace src/CMakeLists.txt \
+      --replace 'if(NOT EXISTS ''${CMAKE_CURRENT_SOURCE_DIR}/dependencies/flann/src)' 'if(FALSE)'
+
+    substituteInPlace src/aliceVision/matching/CMakeLists.txt \
+      --replace ' ''${FLANN_LIBRARY}' ' flann::flann_cpp' \
+      --replace \
+        'alicevision_add_test(matching_test.cpp NAME "matching"          LINKS aliceVision_matching)' \
+        'alicevision_add_test(matching_test.cpp NAME "matching"          LINKS aliceVision_matching lz4)'
+  '';
+
+  # Disable warning causing compile error on certain Clang versions
+  CXXFLAGS = lib.optionalString stdenv.cc.isClang "-Wno-c++11-narrowing";
+
+  cmakeFlags =
+    let
+      cmakeOption = name: enabled: "-D${name}:BOOL=" + (if enabled then "ON" else "OFF");
```

Comment:
Is the ":BOOL" needed?

Reply:
With the `:BOOL` omitted, some of the CMake cache entries are created as strings instead of booleans. Not sure whether it can make a practical difference, though. The build configuration doesn't change here.

Reply:
This is because I am writing a helper function for CMake:

https://github.com/NixOS/nixpkgs/pull/256418

Reply:
For a generic helper function, I would always mandate for setting more exact typing data if it's possible. This way, maintainers could possibly get higher quality metadata for CMake without extra effort, while getting to that point might beconsidered as too tedious at a single package scope.

Reply:
I have found only one place where those "data types" are defined:

https://cmake.org/cmake/help/latest/command/set.html

It looks too gui-driven, and also too dependent on the upstream - what if this variables are truly strings that happen to act like booleans?

Nonetheless, it should be easy to adjust.

Reply:
Any blockers left?

Reply:
Not on my part.

Reply:
Ez, `lib.cmakeBool name value`